### ISSUE TITLE
Can't pick theme files in editor & crash using 'default' in custom theme

### ISSUE THREAD

**Issue by supagu** [godotengine/godot#15593]
I'm using godot 3.0.

Here is the theme:
[default.theme.txt](https://github.com/godotengine/godot/files/1622370/default.theme.txt)
(remove the .txt)

then in the project.godot file set the theme:
theme/custom="res://Theme/default.theme"

You have to do this by hand as the editor only allow you to select a *.tres or *.res file, of which *.theme is likely the only one that should be available here - this is one bug.

Now, this will cause a crash because when you supply "default" in the theme, it looks to take the value from the default theme, which is only created if the custom_theme fails to load, so its looking for defaults in the default theme which has not been instantiated - hence we get a null pointer access. - this is the second (and more show stopping) bug.

Finally, if the theme is incomplete I get a lot of crashes as the default value returned is 0 in the case of a theme value not being configured.

edit: btw, is there a way to get godot to create theme files? I'm having to reverse engineer this myself as I cant seem to find out how.

**Comment by qarmin** [comment]
In Godot  3.2.dev.custom_build. 6d6d43714 I can set to as theme ".theme" file
Provided theme file is incomplete and broken. 
It need a folder "Theme" with "popup_window.png" inside.
It only shows this errors, but doesn't crash editor
```
ERROR: open: Unrecognized binary resource file: res://Theme/default.theme.
   At: core/io/resource_format_binary.cpp:848.
ERROR: load: Condition ' err != OK ' is true. returned: RES()
   At: core/io/resource_loader.cpp:208.
ERROR: _load: Failed loading resource: res://Theme/default.theme.
   At: core/io/resource_loader.cpp:278.
ERROR: load_resource: Condition ' !res.is_valid() ' is true. returned: ERR_CANT_OPEN
   At: editor/editor_node.cpp:632.
```

**Comment by KoBeWi** [comment]
Fixed in 3.2
